typescript/examples/flatbufferswriter/scripts/main.ts

```diff
@@ -0,0 +1,258 @@
+    );
+  },
+  sintancos: ({ x, y, rows, cols, time }: GridDataFuncParams): number => {
+    return (
+      Math.sin(Math.cos(Math.tan((x / cols) * Math.PI + time))) +
+      Math.sin(Math.cos(Math.tan((y / rows) * Math.PI)))
+    );
+  },
+} as { [k: string]: (x: GridDataFuncParams) => number };
+
+function makeNewGrid() {
+  const { cell_size, cell_stride, column_count, row_count, row_stride } = gridParameters;
+
+  const defaultGrid = {
+    timestamp: { sec: 0, nsec: 0 },
+    frame_id: "sensor",
+    pose: {
+      position: {
+        x: -0.5 * cell_size.x * row_count + 2,
+        y: -0.5 * cell_size.y * column_count + 2,
+        z: 0,
+      },
+      orientation: QUAT_IDENTITY,
+    },
+    cell_size,
+    column_count,
+    cell_stride,
+    row_stride,
+    fields: [{ name: "sintancos", offset: 0, type: NumericType.FLOAT32 }],
+  } as Omit<Grid, "data">;
+
+  return defaultGrid;
+}
+
+// adds data to the grid based on the grid parameters and fields
+function addGridData(grid: Omit<Grid, "data">, row_count: number, time: number) {
+  const { column_count, fields, cell_stride, row_stride } = grid;
+  const data = new Uint8Array(row_stride * row_count);
+  const view = new DataView(data.buffer, data.byteOffset, data.byteLength);
+
+  for (let y = 0; y < row_count; y++) {
+    for (let x = 0; x < column_count; x++) {
+      const i = y * row_stride + x * cell_stride;
+      for (const field of fields) {
+        const { name, offset, type } = field;
+        if (type !== NumericType.FLOAT32) {
+          throw new Error("unsupported numeric types");
+        }
+        const dataFunc = fieldDataFuncs[name];
+        if (dataFunc) {
+          const value = dataFunc({
+            x,
+            y,
+            i,
+            rows: row_count,
+            cols: column_count,
+            time,
+          });
+          view.setFloat32(i + offset, value, true);
+        } else {
+          throw new Error(`missing data func for field: ${name}`);
+        }
+      }
+    }
+  }
+
+  (grid as Grid).data = data;
+}
+
+async function addMessageAtTime(
+  mcapFile: McapWriter,
+  gridChannelId: number,
+  getMessageData: (time: number) => Message["data"],
+  time: bigint,
+): Promise<void> {
+  const message = getMessageData(Number(time));
+  await mcapFile.addMessage({
+    channelId: gridChannelId,
+    sequence: 0,
+    publishTime: time,
+    logTime: time,
+    data: message,
+  });
+}
+
+async function main() {
+  const mcapFilePath = "flatbuffer.mcap";
+  const fileHandle = await open(mcapFilePath, "w");
+  const fileHandleWritable = new FileHandleWritable(fileHandle);
+
+  const mcapFile = new McapWriter({
+    writable: fileHandleWritable,
+    useStatistics: false,
+    useChunks: true,
+    useChunkIndex: true,
+  });
+
+  await mcapFile.start({
+    profile: "",
+    library: "mcap example",
+  });
+  const FrameTransformSchemaBuffer = fs.readFileSync(
+    `${__dirname}/../../flatbuffer/bin/FrameTransform.bfbs`,
```

Comment:
Let's get the .bfbs files stored with Git LFS so we don't bloat the repo size too much